align/affineGap_highMem.go

```diff
@@ -25,6 +25,34 @@ func initAffineScoringAndTrace(firstSeqLen int, secondSeqLen int) ([][]int64, []
 	return mRowCurrent, mRowPrevious, mColumn, trace
 }
 
+func initAffineScoringAndTraceRecycle(firstSeqLen int, secondSeqLen int, mRowCurrent, mRowPrevious [][]int64, trace [][][]ColType) ([][]int64, [][]int64, int, [][][]ColType) {
+	mRowCurrent = setSliceSize(mRowCurrent, 3)
+	mRowPrevious = setSliceSize(mRowPrevious, 3)
+	var mColumn int = firstSeqLen + 1
+	trace = setSliceSize(trace, 3)
+	for k := range trace { //k ranges through 3 numbers (0,1,2)
+		trace[k] = setSliceSize(trace[k], firstSeqLen+1)
+		mRowCurrent[k] = setSliceSize(mRowCurrent[k], secondSeqLen+1)
+		mRowPrevious[k] = setSliceSize(mRowPrevious[k], secondSeqLen+1)
+		for i := range trace[0] {
+			trace[k][i] = setSliceSize(trace[k][i], secondSeqLen+1)
+		}
+	}
+	return mRowCurrent, mRowPrevious, mColumn, trace
+}
+
+func setSliceSize[t any](s []t, size int) []t {
```

Comment:
This is neat and might come in handy in other places.  I am not sure of a more general package where we could put it (and people would find it), but maybe someone else has an idea?